drivers/shared/executor/executor_linux.go

```diff
@@ -767,3 +771,53 @@ func cmdMounts(mounts []*drivers.MountConfig) []*lconfigs.Mount {
 
 	return r
 }
+
+// lookupTaskBin finds the file `bin` in taskDir/local, taskDir in that order, then performs
+// a PATH search inside taskDir. It returns an absolute path. See also executor.lookupBin
+func lookupTaskBin(command *ExecCommand) (string, error) {
+	taskDir := command.TaskDir
+	bin := command.Cmd
+
+	// Check in the local directory
+	localDir := filepath.Join(taskDir, allocdir.TaskLocal)
+	local := filepath.Join(localDir, bin)
+	if _, err := os.Stat(local); err == nil {
+		return local, nil
+	}
+
+	// Check at the root of the task's directory
+	root := filepath.Join(taskDir, bin)
+	if _, err := os.Stat(root); err == nil {
+		return root, nil
+	}
+
+	// Find the PATH
+	path := "/usr/local/bin:/usr/bin:/bin"
+	for _, e := range command.Env {
+		if strings.HasPrefix("PATH=", e) {
+			path = e[5:]
+		}
+	}
+
+	// Check the host PATH anchored inside the taskDir
+	if !strings.Contains(bin, "/") {
+		for _, dir := range filepath.SplitList(path) {
+			if dir == "" {
+				// match unix shell behavior, empty path element == .
+				dir = "."
+			}
+			for _, root := range []string{localDir, taskDir} {
```

Comment:
When consulting container PATH, I think we should only look at `taskDir` as it's the root of the container.

Checking `localDir` means that we are interpreting `PATH` differently from the container shell, which isn't ideal.